Improve error message when mpmath is not installed

asmeurer opened the pull request:
Another idea that I hadn't considered would be to include `sympy.mpmath` as a backwards compatible import (that maps to the external mpmath). I think a lot of code uses sympy.mpmath and it would ease transition. I'm not sure how to do it, though.

asmeurer committed:
Don't export print_function and absolute_import

asmeurer committed:
Import mpmath in sympy/__init__.py to give a better error message

moorepants commented on sympy/__init__.py:
```diff
@@ -10,6 +10,15 @@
+    import mpmath
+except ImportError:
+    raise ImportError("SymPy now depends on mpmath as an external library. "
+    "You can install mpmath with 'pip install mpmath'.")
```
How about: "You can install mpmath with 'pip install mpmath' or see the SymPy documentation for other options."

asmeurer replied:
Maybe "see http://docs.sympy.org/latest/install.html for more information" (except that url won't be correct until we release).

moorepants replied:
Sure that's fine too.

certik replied:
Giving a link is fine.

certik commented:
Yep, that was my original proposal as well, but I haven't figured out a way to do that. If you find a way, then we should definitely do it.

asmeurer commented:
I think this does it, but I need to play around with it some more.

```diff
diff --git a/sympy/__init__.py b/sympy/__init__.py
index bc1a34a..a8d1faf 100644
--- a/sympy/__init__.py
+++ b/sympy/__init__.py
@@ -18,8 +18,6 @@
     raise ImportError("SymPy now depends on mpmath as an external library. "
     "You can install mpmath with 'pip install mpmath'.")

-del mpmath
-
 from sympy.release import __version__

 import sys
@@ -29,6 +27,8 @@
     pass
     # Here we can also check for specific Python 3 versions, if needed

+sys.modules['sympy.mpmath'] = mpmath
+
 del sys

```

asmeurer committed:
Provide sympy.mpmath for backwards compatibility

certik commented on sympy/__init__.py:
```diff
@@ -20,6 +27,8 @@
     pass
     # Here we can also check for specific Python 3 versions, if needed
 
+sys.modules['sympy.mpmath'] = mpmath
```
I would not put this here. I would tell users that they can put this into their own code if needed, but sympy itself shouldn't have it.

asmeurer committed:
Recommend to install SymPy through Anaconda

Installing from source is only for advanced users who know what they are
doing.  Only having one recommended installation method makes things less
confusing.

asmeurer committed:
Add mpmath section to the installation docs

asmeurer committed:
Remove sys.modules['sympy.mpmath'] hack

People will have to change their code. Also show in the docs how to do this if
they need to work around dependent code that uses sympy.mpmath.

certik commented on doc/src/install.rst:
```diff
@@ -3,89 +3,58 @@
+========
 
-    $ ./bin/isympy
+`Anaconda <http://continuum.io/downloads>`_ is a free Python a free Python
```
a free Python is there twice.

asmeurer committed:
Fix a typo

certik commented:
+1 to merge as it is now.
